src/ORM/Table.php

```diff
@@ -2178,9 +2178,9 @@ protected function _update(EntityInterface $entity, array $data)
      * any one of the records fails to save due to failed validation or database
      * error.
      *
-     * @param \Cake\Datasource\ResultSetInterface|array<\Cake\Datasource\EntityInterface> $entities Entities to save.
+     * @param \Cake\Datasource\ResultSetInterface<\Cake\Datasource\EntityInterface>|array<\Cake\Datasource\EntityInterface> $entities Entities to save.
```

Comment:
The simplest solution might probably be using `iterable<\Cake\Datasource\EntityInterface>`. The methods related to save/delete many simply loop over the argument anyway and don't use any resultset specific feature.